stream.go

```diff
@@ -107,13 +75,35 @@ type ClientStream interface {
 	// CloseSend closes the send direction of the stream. It closes the stream
 	// when non-nil error is met.
 	CloseSend() error
-	// Stream.SendMsg() may return a non-nil error when something wrong happens sending
-	// the request. The returned error indicates the status of this sending, not the final
-	// status of the RPC.
+	// Context returns the context for this stream.
+	Context() context.Context
+	// SendMsg is generally called by generated code. On error, SendMsg aborts
+	// the stream and returns an io.EOF; the status may be determined by
+	// inspecting the error returned by calling RecvMsg.
+	//
+	// SendMsg blocks until:
+	//   - It schedules m with the transport, or
+	//   - The stream is done, or
+	//   - The stream breaks.
+	//
+	// SendMsg does not wait for an ack. An untimely stream closure can result
+	// in any buffered messages along the way (including those in the
+	// client-side buffer that comes with gRPC by default) being lost. To
+	// ensure delivery, users must call RecvMsg until receiving an EOF before
```

Comment:
This is painfully explicit IMO.  How about:

"SendMsg does not wait until the message is received by the server.  An untimely stream closure may result in lost messages.  To ensure delivery, users should ensure the RPC completed successfully using RecvMsg."

The idea would be to refer people to RecvMsg and make sure we document it well enough there.

Reply:
That is really good. 👍 

Done.